fix: improve deletion

jlabanca opened the pull request:
The branch fixes a bunch of issues with deleting and backspacing in both
natural and financial mode:

* Currently, deleting the decimal in natural mode moves the decimal
portion into the whole number. Now, deleting the decimal move the whole
number into the decimal. See issue #85.
* Pressing backspace in the decimal portion of the number in natural
mode now shifts back to the previous decimal, allowing you to press
backspace multiple times and keep removing numbers.
* Currently, if you select a subset of the numbers and press delete or backspace, nothing happens. Now it will be deleted.
* Currently, if you select most numbers plus the suffix and press delete, the entire number is cleared. This is because the code is comparing selection length to total length without considering suffix length. This is also fixed.
* Currently, pressing backspace before a decimal or comma does nothing. Now it will delete the preceeding number.
* Currently, pressing backspace in the prefix either jumps to the end or to the second digit (depending on the version). Now it will jump to just after the prefix.
* Currently if you press delete before a comma or a decimal, the cursor jumps over it without deleting anything. Now, the cursor jumps over it AND deletes the next digit.
* Also fixes some random issues where the cursor can end up in a place
you don't expect it to.

jlabanca commented:
@dvenkatsagar @nbfontana This PR is ready for review!

jlabanca commented:
@dvenkatsagar - Did you have any specific tests in mind?

Note that the code coverage report is misleading because I added a new `input-handler.spec.ts` file.  The code coverage report claims `input-handler.ts` went from 100% to 28.91% coverage, but in actually went from 0% to 28.91%.  It basically treats a missing test file as 100% coverage.

dvenkatsagar commented:
@jlabanca  Actually the tests do satisfy the changes. I just assumed something is still missing because of the codecov. I was actually good with this change the moment you put in the merge request.

dvenkatsagar commented:
@jlabanca But as you said, ```test/input-handler.spec.ts``` was newly added and currently it only has the tests for the ```handlekeydown``` functionality. 

If we can add the tests for the other events as well like ```handleCut```, ```handleInput```, ```handleKeypress```, and ```handlePaste```, then it should be good.

If you're up for it, please do go ahead and take up this task, else I can do it later after this has been merged..

jlabanca commented:
If you don't mind, I'd prefer to defer the other tests for input-handler.ts until later.  I added tests for the `handlekeydown` method that I modified.  I didn't modify the other methods, so they're no worse off than they were.
